### PR TITLE
Implements geo-restriction and linkless link list proposals: CA-1917

### PR DESCRIPTION
**Pull request by jodieburnett**
Implements the geo-restriction proposal: https://github.com/washingtonpost/ans-schema/blob/master/docs/proposals/2019-10-22-Geo-Restriction-Ids.md

and

Implements the linkless link list proposal: https://github.com/washingtonpost/ans-schema/blob/master/docs/proposals/2019-10-22_Linkless_Link_Lists.md

### CODE REVIEW

File: src/main/resources/schema/ans/0.10.4/traits/trait_content_restrictions.json [hunk 1]
```diff
@@ -29,6 +29,29 @@
         }
       },
       "required": [ "active" ]
+    },
+    "geo": {
+      "type": "object",
+      "description": "Geo-Restriction configuration that contains the restriction ids that this content should be associated with.",
+      "additionalProperties": false,
+      "properties": {
+        "restrictions": {
```
Comment (gengel) [inline src/main/resources/schema/ans/0.10.4/traits/trait_content_restrictions.json]: To minimize the number of ways to "null" this field out, I'd suggest:
* Making `restrictions` a required property of geo
* Making the `minItems` the same as the `maxItems`: 1

That way there's only one way to indicate that a story has no restrictions, and that's by omitting `geo` entirely. (I.e. all saved stories today.)

We can relax these restrictions later if a use case arrives for a `geo` trait with a field other than `restrictions`

File: src/main/resources/schema/ans/0.10.4/traits/trait_content_restrictions.json [hunk 2]
```diff
@@ -29,6 +29,29 @@
...
+          "type": "array",
+          "description": "An array containing the geo-restriction objects. Limited to a size of 1 for now.",
+          "maxItems": 1,
+          "items": {
+            "type": "object",
+            "description": "An object specifying the _id of the restriction this content should be associated with.",
+            "additionalProperties": false,
+            "properties": {
+              "restriction_id": {
```
Comment (gengel) [inline src/main/resources/schema/ans/0.10.4/traits/trait_content_restrictions.json]: Similarly let's make `restriction_id` required

File: tests/fixtures/schema/0.10.4/link-list-fixture-good-text.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+  "_id": "ABC",
+  "type": "link_list",
+  "subtype": "more-information",
+  "alignment": "right",
+
```
Comment (anscho) [inline tests/fixtures/schema/0.10.4/link-list-fixture-good-text.json]:
```suggestion
```
Remove line break

File: tests/fixtures/schema/0.10.4/link-list-fixture-good-text.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  "title": "Learn More",
+  "items": [
+    {
+      "_id": "GHI",
+      "type": "interstitial_link",
+      "subtype": "skinny",
+
```
Comment (anscho) [inline tests/fixtures/schema/0.10.4/link-list-fixture-good-text.json]:
```suggestion
```
Remove line break

File: tests/fixtures/schema/0.10.4/link-list-fixture-good-text.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      "url": "https://www.washingtonpost.com",
+      "content": "See more from The Washington Post"
+    },
+
```
Comment (anscho) [inline tests/fixtures/schema/0.10.4/link-list-fixture-good-text.json]:
```suggestion
```
Remove line break

File: tests/fixtures/schema/0.10.4/video-fixture-bad-empty-geo-restriction.json [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    "foo": "bar"
+  },
+  "corrections": [
+    {
+      "type": "correction",
+      "correction_type": "clarification",
+      "text": "An earlier version of this video stated that Dewey defeated Truman. Truman defeated Dewey."
+    }
+  ]
+
```
Comment (anscho) [inline tests/fixtures/schema/0.10.4/video-fixture-bad-empty-geo-restriction.json]:
```suggestion
```
Remove line break

File: tests/fixtures/schema/0.10.4/video-fixture-bad-empty-geo-restriction.json [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+  "subtitles": {
+    "confidence": 0.657583939,
+    "subtitlesUrls": [{
+      "format": "DFXP",
+      "url": "https://closedcaptions.posttv.com/06-30-2015/5592c09fe4b082c8417f3f72_5592c013e4b082c8417f3f71/t_1435683757146.dfxp"
+    }, {
+      "format": "RAW_TEXT",
+      "url": "https://closedcaptions.posttv.com/06-30-2015/5592c09fe4b082c8417f3f72_5592c013e4b082c8417f3f71/t_1435683757146.rawtext"
+    }
+                     ]
```
Comment (anscho) [inline tests/fixtures/schema/0.10.4/video-fixture-bad-empty-geo-restriction.json]:
```suggestion
```

File: tests/fixtures/schema/0.10.4/video-fixture-bad-empty-geo-restriction.json [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+  },
+  "subtitles": {
+    "confidence": 0.657583939,
+    "subtitlesUrls": [{
+      "format": "DFXP",
+      "url": "https://closedcaptions.posttv.com/06-30-2015/5592c09fe4b082c8417f3f72_5592c013e4b082c8417f3f71/t_1435683757146.dfxp"
+    }, {
+      "format": "RAW_TEXT",
+      "url": "https://closedcaptions.posttv.com/06-30-2015/5592c09fe4b082c8417f3f72_5592c013e4b082c8417f3f71/t_1435683757146.rawtext"
+    }
```
Comment (anscho) [inline tests/fixtures/schema/0.10.4/video-fixture-bad-empty-geo-restriction.json]:
```suggestion
    }]
```

File: tests/schema-tests-010.js [hunk 9]
```diff
@@ -1065,7 +1089,11 @@ describe("Schema: ", function() {
               validateIfFixtureExists(version, type_prefix + '/link_list.json', 'link-list-fixture-good');
             });
 
-            it("should not validate a non-link-list", function () {
+            it("should validate a link list with a text item", function () {
```
Comment (anscho) [inline tests/schema-tests-010.js]: It could be worth adding a test with all/only text items.

**Pull request merged**